### ISSUE TITLE
Getting 404 errors

### ISSUE THREAD

**Issue by dportnoy** [usagov/Federal-Agency-Directory-API-Documentation#3]
Looks like even the example URLs are getting 404 errors:
- https://www.usa.gov/api/USAGovAPI/contacts.json/contact/49021
- https://www.usa.gov/api/USAGovAPI/contacts.json/contact/49021/tree/descendant
  ![image](https://cloud.githubusercontent.com/assets/2925801/19096377/87c58d4c-8a61-11e6-881b-25eb0182973f.png)

**Comment by 24AheadDotCom** [comment]
I sent an email to usagov-developers _at_ gsa.gov about this 6 days ago and I haven't heard back. I'm going to tweet them and see if that works.

**Comment by cooperfellows** [comment]
Seeing the same issue in a project I had been using this data. Did you ever get an update?

**Comment by MayankSingla13** [comment]
I am getting this error too. I sent an email but haven't heard back. I need list of all agencies and their parent agencies so that I can create a hierarchy for a project.

**Comment by thedore17** [comment]
Anyone ever get any resolution to this?

**Comment by MayankSingla13** [comment]
No man. I have sent multiple emails but never got a reply.

**Comment by thedore17** [comment]
Wow. Did you find an alternate source for the data or did you just give up? I'm mostly looking for the hierarchy in machine readable format. Ideally with contact email and office addresses too. I thought I hit the goldmine with this thing.  :(

**Comment by cooperfellows** [comment]
I have a static list from about a year ago with this information. I can
share it as a Google.doc next week when I'm working again if you want.

**Comment by MayankSingla13** [comment]
I used this: https://www.federalregister.gov/developers/api/v1/
The federal register is old. Some agencies don't exist anymore but they are listed in there. Plus, new agencies have been added in the govt - they are not listed as well.

Hey @cooperfellows can you please share that?

**Comment by thedore17** [comment]
Hey @cooperfellows -- definitely would be great if you're willing to share that. @MayankSingla13 Thanks for the Federal Register link. It would work for me if it had address or email fields, but it doesn't have either. There is a main website field that I might be able to regex email domains out of, but my god that doesn't sound fun.

What I find really frustrating is that clearly this API still exists and functions because their own website is clearly using it. That's our government for you.... 

Thanks for your help guys.

**Comment by cooperfellows** [comment]
Hey @thedore17 attached is a static list from around Mid September of last year. It's not the complete set of data for each agency, but it's a good chunk of it. I want to be clear that this data could be out of date/incorrect, as I found the API to be lacking in accurate data now and then, but its certainly better than nothing. 
[Federal Agencies via API as of 2016-09-14.xlsx](https://github.com/usagov/Federal-Agency-Directory-API-Documentation/files/758155/Federal.Agencies.via.API.as.of.2016-09-14.xlsx)

**Comment by santiago-afonso** [comment]
@cooperfellows I cannot, just cannot believe the luck that I just came to this place and you posted that file. Made my day. Thanks!

**Comment by cooperfellows** [comment]
@osnofas Glad you will find it helpful, but again, I take no responsibility for the accuracy of all that data.

It's what the API returned in mid-September of 2016, and as noted, the API was not always accurate.

**Comment by santiago-afonso** [comment]
I see what you mean - apparently it thinks that Washington is in Arkansas.

**Comment by cooperfellows** [comment]
Yep, we wound up using "Zip" instead of the "Statter" field because AK (which is Alaska) kept showing up for Agencies in Washington DC. The Zip field tended to be more reliable, when it was available, because not every record had one :), depending on your use case, you could use a geo lookup API to get the correct state based on that Zip.

My guess is that "AK" is the default value for the "Stateter" field in whatever system people used when entering this data into the API (AK is the first state if abbreviations are listed alphabetically)

**Comment by thedore17** [comment]
Could you use the full address and geo-lookup that way? Or are the address
records spotty too?

**Comment by cooperfellows** [comment]
I didn't work with the data too much myself, handed it off to someone else on the team for import into another system. The "stateter" and "Fed Branch" columns provided the most trouble for him from what I could dig up. Not sure about the Address field. It wasn't a big enough problem that I heard about it, so I guess that's good?

**Comment by thedore17** [comment]
Hey Folks, I went through the doc kindly provided by @cooperfellows to clean up all the errant AK's. Interestingly enough there was only one accurate AK out of the entire 474 office list. [I am sharing it as a Google Doc here](https://docs.google.com/spreadsheets/d/1F2WjJfYeZJWmL6FvO-IHGSN95S77YLdZ3ByKTihfnAo/edit?usp=sharing).  I also made this data more hacker friendly by converting it to JSON. Please let me know if you find any issues: https://github.com/thedore17/FederalData/blob/master/FederalAgencies-2016-9-14.json

**Comment by thedore17** [comment]
Quick update: Combing through the data has yielded other data anomalies and inconsistencies. Many of the address 1 fields are messed up, so I am fixing those to the best of my abilities while maintaining as much of the original data as I can.

**Comment by carltoncolter** [comment]
A year later and I am getting the same errors.  Was it taken down or is it experiencing a problem?

**Comment by thedore17** [comment]
It was shut down. I got what data I could, and wrote some data massagers in
order to improve the quality. There are still many holes though:
https://github.com/thedore17/FederalData
